packages/lifi-adapter/src/service.ts

```diff
@@ -0,0 +1,367 @@
+  ): RateType {
+    const timestamp = new Date().toISOString();
+
+    try {
+      const sourceDecimals = Number.isFinite(route.source.decimals) ? route.source.decimals : 0;
+      const destinationDecimals = Number.isFinite(route.destination.decimals) ? route.destination.decimals : 0;
+
+      const normalizedIn = new Decimal(notional).div(new Decimal(10).pow(sourceDecimals));
+
+      if (normalizedIn.isZero()) {
+        throw new Error('Fallback notional resolved to zero');
+      }
+
+      const normalizedOut = normalizedIn;
+      const rawOut = normalizedOut.mul(new Decimal(10).pow(destinationDecimals));
+
+      return {
+        source: route.source,
+        destination: route.destination,
+        amountIn: notional,
+        amountOut: rawOut.toFixed(0, Decimal.ROUND_DOWN),
+        effectiveRate: normalizedOut.div(normalizedIn).toNumber(),
+  totalFeesUsd: 0,
+        quotedAt: timestamp,
+      } satisfies RateType;
+    } catch {
+      return {
+        source: route.source,
+        destination: route.destination,
+        amountIn: notional,
+        amountOut: notional,
+        effectiveRate: 1,
+  totalFeesUsd: 0,
+        quotedAt: timestamp,
+      } satisfies RateType;
+    }
+  }
+
+  /**
+   * Get complete snapshot of provider data for given routes and notionals.
+   *
+   * This method coordinates fetching:
+   * - Volume metrics for specified time windows
+   * - Rate quotes for each route/notional combination
+   * - Liquidity depth at 50bps and 100bps thresholds
+   * - List of supported assets
+   */
+  getSnapshot(params: {
+    routes: Array<{ source: AssetType; destination: AssetType }>;
+    notionals: string[];
+    includeWindows?: Array<"24h" | "7d" | "30d">;
+  }) {
+    if (!params?.routes?.length || !params?.notionals?.length) {
+      return Effect.fail(new Error('Routes and notionals are required'));
+    }
+
+    return Effect.tryPromise({
+      try: async () => {
+        try {
+          const [volumes, rates, liquidity, listedAssets] = await Promise.all([
+            this.getVolumes(params.includeWindows || ["24h"]),
+            this.getRates(params.routes, params.notionals),
+            this.getLiquidityDepth(params.routes),
+            this.getListedAssets()
+          ]);
+
+          return {
+            volumes,
+            rates,
+            liquidity,
+            listedAssets,
+          } satisfies ProviderSnapshotType;
+        } catch (error) {
+          throw new Error(`Snapshot fetch failed: ${error instanceof Error ? error.message : String(error)}`);
+        }
+      },
+      catch: (error: unknown) =>
+        new Error(`Failed to fetch snapshot: ${error instanceof Error ? error.message : String(error)}`)
+    });
+  }
+
+  /**
+   * Li.Fi doesn't provide volume metrics, return empty array
+   */
+  private async getVolumes(_windows: Array<"24h" | "7d" | "30d">): Promise<VolumeWindowType[]> {
+    // Li.Fi does not expose aggregated volume metrics via its public API.

```

Comment:
Taking note that Li.Fi does not expose volume metrics.

Any links to docs? How might we work around this?